Sources/SpeziFHIR/FHIRStore.swift

```diff
@@ -19,149 +17,151 @@ import Spezi
 /// The ``FHIRStore`` is automatically injected in the environment if you use the ``FHIR`` standard or can be used as a standalone module.
 @Observable
 public final class FHIRStore: Module,
-                              EnvironmentAccessible,
-                              DefaultInitializable,
-                              @unchecked Sendable /* `unchecked` `Sendable` conformance fine as access to `_resources` protected by `NSLock` */ {
-    private let lock = NSLock()
-    @ObservationIgnored private var _resources: [FHIRResource]
-    
-    
-    /// Allergy intolerances.
+                             EnvironmentAccessible,
+                             DefaultInitializable,
+                             Sendable {
+    /// Actor-isolated storage for `FHIRResource`s backing the ``FHIRStore``.
+    private actor Storage {
```

Comment:
Could `Storage` be part of a private `FHIRStore` extension? This is of course a coding style and I am quite okay the way that it is, but I find it a bit more clean like this.

Reply:
For sure, we can definitely go for something like that!